var/spack/repos/builtin/packages/py-torch/package.py

```diff
@@ -156,7 +156,8 @@ class PyTorch(PythonPackage, CudaPackage):
     depends_on('cudnn@6:7', when='@:1.0+cudnn')
     depends_on('cudnn@7.0:7', when='@1.1:1.5+cudnn')
     depends_on('cudnn@7:', when='@1.6:+cudnn')
-    depends_on('magma', when='+magma')
+    depends_on('magma+cuda~rocm', when='+magma+cuda')
+    depends_on('magma+rocm~cuda', when='+magma+rocm')
```

Comment:
cuda and rocm are mutually exclusive, so you can simplify this:
```suggestion
    depends_on('magma+cuda', when='+magma+cuda')
    depends_on('magma+rocm', when='+magma+rocm')
```

Reply:
I thought so as well, but could swear Spack tried to build rocm on Lassen, despite it not really being appropriate.  I added the extra explicit removal of those conflicting variants just to be sure.

Reply:
If you can reproduce that, open a bug report. It might choose the wrong default, but it shouldn't be possible to concretize `+rocm+cuda` for magma or PyTorch